lib/arelastic/nodes/node.rb

```diff
@@ -22,6 +24,26 @@ def read_option!(options, key)
       def ==(other)
         other.is_a?(Arelastic::Nodes::Node) && as_elastic == other.as_elastic
       end
+
+      def hash
+        as_elastic.hash
+      end
+
+      def eql?(other)
+        as_elastic.eql?(other.as_elastic)
```

Comment:
Could this alias `==`?

[Object doc](https://ruby-doc.org/core-3.1.2/Object.html#method-i-eql-3F):

> For objects of class [Object](https://ruby-doc.org/core-3.1.2/Object.html), [eql?](https://ruby-doc.org/core-3.1.2/Object.html#method-i-eql-3F) is synonymous with #==. Subclasses normally continue this tradition by aliasing eql? to their overridden #== method, but there are exceptions.

Reply:
Fixed